lib/rails_core_extensions/translations.rb

```diff
@@ -0,0 +1,44 @@
+module RailsCoreExtensions
+  module Translations
+    def self.included(base)
+      base.extend ClassMethods
+    end
+
+    module ClassMethods
+      def translate(key, options = {})
+        I18n.translate key, options.merge(scope: translation_key)
+      end
+
+      def translation_key
+        (base_activerecord_class || base_ruby_class).name.tableize.singularize
+      end
+
+      def base_activerecord_class
+        return unless defined?(ActiveRecord) && is_a?(ActiveRecord::Base)
+        klass = self
+        while klass.superclass != ActiveRecord::Base
+          klass = klass.superclass
+        end
+        klass
+      end
+
+      def base_ruby_class
+        klass = self
+        while klass.superclass != Object
+          klass = klass.superclass
+        end
+        klass
+      end
+
+      def t(key, options = {})
+        self.translate(key, options)
+      end
+    end
+
+    private
+
+    def t(key, options = {})
```

Comment:
Yes, and when I thought about extracting it I was tempted to do i18n_extensions gem to highlight how independent this code is.

Ideally I would think active_record, etc. could go into their own extensions gems, but too much effort.